sway/tree/container.c

```diff
@@ -1520,5 +1520,9 @@ void container_raise_floating(struct sway_container *con) {
 }
 
 bool container_is_scratchpad_hidden(struct sway_container *con) {
-	return con->scratchpad && !con->workspace;
+	struct sway_container *target = con;
+	while (target->parent) {
```

Comment:
This change breaks frame updates when running `fullscreen global` on a scratchpad hidden container